pysd/py_backend/xmile/SMILE2Py.py

```diff
@@ -125,7 +98,28 @@
     "mod": "%",
 }
 
-builders = {}
+# ====
+# 3.5.3 Delay Functions
+# http://docs.oasis-open.org/xmile/xmile/v1.0/csprd01/xmile-v1.0-csprd01.html#_Toc398039982
+# ====
+
+builders = {
+    # !TODO! Should correct handle the last args, should take the initial value of element instead the 0 if this parameter is missing
+    # "delay" !TODO! How to add the infinity delay?
+    "delay1": lambda element, subscript_dict, args: builder.add_n_delay(args[0], args[1], args[2] if len(args) > 2 else 0, "1", element['subs'], subscript_dict),
```

Comment:
This is an interesting way of doing it. It sounds like xmile has an optional initialization argument, whereas vensim uses the smooth3I syntax? In the Vensim syntax for this function, there are no optional arguments, so we can make it more explicit, which is nice. I do like that you make the element and the subscript dict explicit parameters to the lambda, that's probably good practice that should make its way to vensim2py as well. =)

One issue is that after the else, it should probably be another call to the input variable. I've just discovered this bug in the vensim2py version and haven't dealt with it yet. See: https://github.com/JamesPHoughton/pysd/issues/145 Does the xmile spec specify how the outflow should be initialized? The vensim behavior is not what I originally thought. I've updated the vensim model tests in the test repo to raise this error, but not the .xmile files.

Reply:
I'm looks to vensim2py for examples :)
But you initialize builders configurations every parse method, this way allow initialize it only once.

> Does the xmile spec specify how the outflow should be initialized?
Yes, if `initial value` is not sets, that should takes the initial value of input component. It described in specification. I'm already saw the #145 issue and leave this open for xmile. And in the next iterations try to solve it.

Reply:
Sounds good.